api/entity_operations.go

```diff
@@ -33,6 +33,117 @@ func (c *Client) CreateGovernmentNode() (*models.Entity, error) {
 	return createdEntity, nil
 }
 
+// GetPresidentByGovernment retrieves a president entity (citizen with AS_PRESIDENT relationship to government) by name
+func (c *Client) GetPresidentByGovernment(presidentName string) (*models.Entity, error) {
+	// Get the president entity ID - presidents are citizens with AS_PRESIDENT relationship to government
+	presidentResults, err := c.SearchEntities(&models.SearchCriteria{
+		Kind: &models.Kind{
+			Major: "Person",
+			Minor: "citizen",
+		},
+		Name: presidentName,
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to search for president entity: %w", err)
+	}
+	if len(presidentResults) == 0 {
+		return nil, fmt.Errorf("president entity not found: %s", presidentName)
+	}
+
+	// Find the president by checking if they have AS_PRESIDENT relationship to government
+	for _, president := range presidentResults {
+		// Get government node
+		governmentResults, err := c.SearchEntities(&models.SearchCriteria{
+			Kind: &models.Kind{
+				Major: "Organisation",
+				Minor: "government",
+			},
+		})
+		if err != nil || len(governmentResults) == 0 {
+			continue
+		}
+
+		// Check if this citizen has AS_PRESIDENT relationship to government
+		presidentRelations, err := c.GetRelatedEntities(governmentResults[0].ID, &models.Relationship{
+			Name:            "AS_PRESIDENT",
+			RelatedEntityID: president.ID,
+		})
+		if err != nil {
+			continue
+		}
+
+		// If there are any AS_PRESIDENT relationships (active or not), return the president
+		if len(presidentRelations) > 0 {
+			// Convert SearchResult to Entity
+			entity := &models.Entity{
+				ID:         president.ID,
+				Kind:       president.Kind,
+				Created:    president.Created,
+				Terminated: president.Terminated,
+				Name: models.TimeBasedValue{
+					Value: president.Name,
+				},
+				Metadata:      []models.MetadataEntry{},
+				Attributes:    []models.AttributeEntry{},
+				Relationships: []models.RelationshipEntry{},
+			}
+			return entity, nil
+		}
+	}
+
+	return nil, fmt.Errorf("president entity not found or not active: %s", presidentName)
+}
+
+// GetMinisterByPresident retrieves a minister entity by president name and minister name
+func (c *Client) GetMinisterByPresident(presidentName, ministerName, dateISO string) (*models.Entity, error) {
+	// Get the president entity using the helper function
+	presidentEntity, err := c.GetPresidentByGovernment(presidentName)
+	if err != nil {
+		return nil, err
+	}
+	presidentID := presidentEntity.ID
+
+	// Get all minister relationships for the president
+	presidentRelations, err := c.GetRelatedEntities(presidentID, &models.Relationship{
+		Name: "AS_MINISTER",
```

Comment:
@zaeema-n let's not use strings here. Use a constant value or an Enum. Because this is error prone.